components/esp_modem/examples/modem_tcp_client/components/extra_tcp_transports/tls_transport.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "esp_log.h"
+#include "esp_transport.h"
+#include "mbedtls_wrap.hpp"
+
+static const char *TAG = "tls_transport";
+
+class TlsTransport: public Tls {
+public:
+    explicit TlsTransport(esp_transport_handle_t parent) : Tls(), transport_(parent) {}
+    int send(const unsigned char *buf, size_t len) override;
+    int recv(unsigned char *buf, size_t len) override;
+    static bool set_func(esp_transport_handle_t tls_transport);
+
+private:
+    esp_transport_handle_t transport_{};
+    int connect(const char *host, int port, int timeout_ms);
+
+    struct priv {
```

Comment:
Could you elaborate the reason to group the functions in the internal struct?

Reply:
Two reasons:
1) all these functions belong together (they basically define one transport)
2) namespacing: we already use generic names like `connect()` or `send()` in the class. This helps using the same here and in tcp-transport

Reply:
Ok, I suggest renaming the struct to `transport` or even `tcp_transport` there should be no namespacing issue since the functions are already scoped by the class, but I this the inner struct is still a good idea.

Reply:
Have renamed to `transport`.

>  but I this the inner struct is still a good idea.

I can also prefix with `transport_xxx` (c style namespacing) if this looks better for you? For example:

`TlsTransport::transport::read()` -> `TlsTransport::transport_read`